### PR TITLE
Bump dependencies / ESM / bundle with esbuild

### LINKED ISSUE #34: `set-output` is deprecated
I received the following warning when I was running the action.

> Warning: The `set-output` command is deprecated and will be disabled soon. Please upgrade to using Environment Files. For more information see: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/

### PR DESCRIPTION
**Pull request by CarsonF**
- Run on Node v12 -> v20
- Bundle with `esbuild` instead of `@vercel/ncc`
  Never even heard of ncc until this, not really sure their purpose.
  `esbuild` is definitely _one of_ the new standards. 
- Bump all libraries (solves legacy `@actions/core` warnings #34)
- Opted into more ESM.
  Though the build is still CJS, as that's what `@actions/core` emits for now.
  Some work went into mocking ESM modules for Jest, but I believe the result is a nicer experience anyways.
- Bump some of the edgedb package versions referenced in tests to more recent ones
- Support (forward) arm64 (aarch64) architecture

Fixes #34 

I really tried to limit any logical changes here, since this is so large in other ways.
I would like to refactor further in later PRs if allowed.

### CODE REVIEW

**Comment by elprans** [comment]
Tests seem to be failing with

```
ReferenceError: require is not defined in ES module scope, you can use import instead
This file is being treated as an ES module because it has a '.js' file extension and '/home/runner/work/setup-edgedb/setup-edgedb/package.json' contains "type": "module". To treat it as a CommonJS script, rename it to use the '.cjs' file extension.
```

**Comment by CarsonF** [comment]
Thanks for approving CI. I had a hard time with the test suite since there's not a binary for M1/ARM chips.
Let me address CI failures.

**Comment by elprans** [comment]
> there's not a binary for M1/ARM chips

Which binary?  Both server and CLI have native m1 and linux aarch64 builds (though possibly not ancient releases that might be referred to in the tests, so try using a more recent server/CLI version there).

**Comment by CarsonF** [comment]
>   ● setup-edgedb › Installs CLI

    This action does not support the arm64 architecture

      236 |     distArch = 'x86_64'
      237 |   } else {
    > 238 |     throw Error(`This action does not support the ${arch} architecture`)
          |           ^
      239 |   }
      240 |
      241 |   return `${distArch}-${distPlatform}`

      at Module.getBaseDist (src/main.ts:238:11)
      at Object.<anonymous> (__tests__/setup.test.ts:79:27)

https://github.com/edgedb/setup-edgedb/blob/main/src/main.ts#L230
https://github.com/edgedb/setup-edgedb/blob/main/__tests__/setup.test.ts#L92

**Comment by elprans** [comment]
Oh, feel free to add `arm64`, normalizing to `aarch64`, it should work fine

**Comment by CarsonF** [comment]
@elprans I think I've fixed the problems. Trigger CI for me?

**Comment by CarsonF** [comment]
Thanks, will continue

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -67,7 +67,7 @@ jobs:
       fail-fast: false
     services:
       edgedb:
-        image: edgedb/edgedb:1-rc2
+        image: edgedb/edgedb:1
```
Comment (elprans) [inline .github/workflows/test.yml:70]:
```suggestion
        image: edgedb/edgedb:3
```

**Comment by CarsonF** [comment]
Everything looks good here. Ready for review/merge when able 😄

**Review by elprans — APPROVED** [review]
LGTM! Thanks for working on this!

**Pull request merged**